app/services/ingest_ephemera_csv.rb

```diff
@@ -104,22 +103,44 @@ def language
     @language ||= vocab_service.find_term(label: ISO_639.find_by_code(fields[:language]).english_name.split(";").first).id
   end
 
-  def geo_origin
-    return unless fields[:geo_origin].present?
-    @geo_origin ||= vocab_service.find_term(label: fields[:geo_origin]).id
+  def geographic_origin
+    return unless fields[:geographic_origin].present?
+    @geographic_origin ||= vocab_service.find_term(label: fields[:geographic_origin])
+  end
+
+  def keywords
+    return unless fields[:keywords].present?
+    fields[:keywords].split(",")
   end
 
   def subject
-    return unless fields[:subjects].present?
-    subjects = fields[:subjects].split("/").map { |s| s.split("--") }.map { |c, s| { "category" => c, "topic" => s } }
+    return unless fields[:subject].present?
+    subjects = fields[:subject].split("/").map { |s| s.split("--") }.map { |c, s| { "category" => c, "topic" => s } }
     subjects.uniq.map do |sub|
-      vocab_service.find_subject_by(category: sub["category"], topic: sub["topic"]).id
+      subject = vocab_service.find_subject_by(category: sub["category"], topic: sub["topic"])
+      subject&.id
     end
   end
 
   def geo_subject
     return unless fields[:geo_subject].present?
-    Array(vocab_service.find_term(label: Array(fields[:geo_subject]).first, vocab: "LAE Areas"))
+    Array(vocab_service.find_term(label: Array(fields[:geo_subject]).first))
```

Comment:
I think what'll happen here is this'll look right in the UI and everything, but in the edit form unless the project's geographic_subject is associated with whatever vocabulary this happens to find first, it won't be selectable or show up from the dropdown.

Reply:
This seemed like a hack to me, to hard-code the vocabulary for a particular project here, though I've come to understand that "LAE Areas" is actually a general-purpose vocabulary. 

I'll restore it.